sdk/rust-examples/huffman-encoding/src/main.rs

```diff
@@ -0,0 +1,161 @@
+//! Huffman Encoding Example
+//!
+//! ## Context
+//!
+//! Reads user input, and then encodes the file to binary format to compress it and save space.
+//! And then, it prints the encoded file.
+//! After that, the encoded file gets converted back to the original text, then that is outputted.
+//! WARNING: Program is only desinged for ASCII characters, not UTF-8 encoding !!
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use anyhow::{self, Ok};
+use std::{collections::HashMap, fs};
+
+const INPUT_FILENAME: &'static str = "/input/huffman_example_input.txt";
+const OUTPUT_FILENAME: &'static str = "/output/encoded_output.txt";
+
+type Link = Option<Box<Node>>;
+
+#[derive(Debug)]
+
+/// A Binary Tree Node is represented here
+struct Node {
+    /// Each Node, must represent a character, and its frequency in the input string
+    freq: i32,
+    ch: Option<char>,
+
+    /// The nodes for left and right children of the node
+    left: Link,
+    right: Link,
+}
+
+/// function to create a new node, (Like a Constructor function)
+fn new_node(freq: i32, ch: Option<char>) -> Node {
```

Comment:
You can mark small functions like this as `inline` to ensure that the Rust compiler inlines them during compilation, like so:

```rust
#[inline]
fn new_node(freq: i32, ch: Option<char>) -> Node {
   ...
}
```

Reply:
I have encoded all functions except assign_codes() since it's a recursive function. Is this correct?

Reply:
I would only mark `new_node` as inline as it is essentially a function whose body is a single expression (i.e., creating and returning a new `Node`).  Anything more complicated than that I let the compiler decide whether to inline...

Reply:
I'll rectify this. And hopefully, the code will get merged this time🤞